client/client.go

```diff
@@ -12,34 +12,49 @@ import (
 )
 
 var (
-	app   = kingpin.New("client", "send messages to a server")
-	host  = app.Flag("host", "host to connect to").Required().String()
-	send  = app.Flag("message", "send a request to server").Required().Strings()
-	token = app.Flag("token", "authentication token").Required().String()
+	app = kingpin.New("client", "connect to server")
+
+	send    = app.Command("send", "send message to server")
+	host    = send.Flag("host", "host to connect to").Required().String()
+	message = send.Flag("message", "send a request to server").Required().Strings()
+	token   = app.Flag("token", "authentication token").Required().String()
 )
 
 func main() {
-	kingpin.MustParse(app.Parse(os.Args[1:]))
-	message := strings.Join(*send, " ")
-	reader := strings.NewReader(message)
-
-	req, err := http.NewRequest("POST", *host+"/echo", reader)
-
-	// add token header to the req
-	req.Header.Add("Token", *token)
-
-	client := &http.Client{}
-	resp, err := client.Do(req)
-
-	if err != nil {
-		log.Println(err)
-	}
-
-	defer resp.Body.Close()
-	bodyBytes, err := ioutil.ReadAll(resp.Body)
-	serverOutput := string(bodyBytes)
-	if serverOutput != message {
-		fmt.Println(*host + " did not echo your request, oops! got:")
+	switch kingpin.MustParse(app.Parse(os.Args[1:])) {
+
+	case send.FullCommand():
+		requestMessage := strings.Join(*message, " ")
+		reader := strings.NewReader(requestMessage)
+		endpoint := *host + "/echo"
+		req, err := http.NewRequest("POST", endpoint, reader)
+		if err != nil {
+			fmt.Println("request could not be initiated: ", err)
+			return 
```

Comment:
Do you want to return a non-zero exit code from your client in this case? Here is the behavior of some other client utilities:

```
walt@work:~/git/teleport$ curl http://nope.gravitational.com           
curl: (6) Could not resolve host: nope.gravitational.com
walt@work:~/git/teleport$ echo $?
6
walt@work:~/git/teleport$ wget http://nope.gravitational.com
--2020-09-24 10:01:59--  http://nope.gravitational.com/
Resolving nope.gravitational.com (nope.gravitational.com)... failed: Name or service not known.
wget: unable to resolve host address ‘nope.gravitational.com’
walt@work:~/git/teleport$ echo $?
4
walt@work:~/git/teleport$ curl http://gravitational.com > /dev/null
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
walt@work:~/git/teleport$ echo $?
0
```

Same remark could apply to several other of the returns in this main.